storage/innobase/include/trx0purge.h

```diff
@@ -243,22 +231,25 @@ class purge_sys_t
 					record */
 	uint16_t	hdr_offset;	/*!< Header byte offset on the page */
 
-  /** Binary min-heap of indexes in trx_sys.rseg_array, ordered on
-  rseg_t::last_trx_no(). It is protected by the pq_mutex */
+  /** Binary min-heap of (trx_no, trx_sys.rseg_array index) pairs, ordered on
+  trx_no. The highest 56 bits of each element is trx_no, the lowerst 8 bits -
+  index in trx_sys.rseg_array. It is protected by the pq_mutex */
   purge_queue purge_queue;
```

Comment:
I would omit the implementation detail about the `container_type::value_type` layout at this level.